lib/AST/ASTMangler.cpp

```diff
@@ -1228,28 +1223,54 @@ void ASTMangler::appendProtocolName(const ProtocolDecl *protocol) {
   appendDeclName(protocol);
 }
 
-void ASTMangler::appendNominalType(const NominalTypeDecl *decl) {
+void ASTMangler::appendAnyGenericType(const GenericTypeDecl *decl) {
   // Check for certain standard types.
   if (tryAppendStandardSubstitution(decl))
     return;
 
   // For generic types, this uses the unbound type.
-  TypeBase *key = decl->getDeclaredType().getPointer();
+  Type key;
+  if (auto *alias = dyn_cast<TypeAliasDecl>(decl)) {
```

Comment:
Fine with me. Does it return UnboundGenericType, or just Type to handle the non-generic case?